lib/std/fs.zig

```diff
@@ -443,7 +443,7 @@ pub fn symLinkAbsoluteW(
 ) !void {
     assert(path.isAbsoluteWindowsWTF16(target_path_w));
     assert(path.isAbsoluteWindowsWTF16(sym_link_path_w));
-    return windows.CreateSymbolicLink(null, sym_link_path_w, target_path_w, flags.is_directory);
+    return windows.CreateSymbolicLink(null, sym_link_path_w, @as([:0]const u16, @ptrCast(target_path_w)), flags.is_directory);
```

Comment:
Is there a function to go from `[]const T -> [:0]const T` ?

Reply:
I guess I can just do `@ptrCast(target_path_w)` and the inferred result is [:0]const u16

Reply:
In general, casting to zero-terminated only works if there is actually a zero terminator in memory. If you can't guarantee this, then you'd need to copy to a buffer large enough and set the terminator. Exception is if you know you're calling a system API that accepts length and pointer which most modern Windows APIs do, and pass the correct length with the pointer.

Reply:
Agreed this should be the case for native Zig API. In the case of wrapping Windows functions, Windows APIs, because they are a C interface, do not have a means to declare a zero-terminated string in the type system (to some extent, naming conventions and tooling like SAL have been used to attempt to deal with this hole in C).

Reply:
> In my opinion however, the function signature should express that a 0 terminated string is required.

Right. Yeah I wasn't sure if changing the function signature was ok. To clarify do you mean you'd like `symLinkAbsoluteW` to change its parameters from `[]const u16` to `[:0]const u16` ?

```zig
pub fn symLinkAbsoluteW(
    target_path_w: []const u16,
    sym_link_path_w: []const u16,
    flags: Dir.SymLinkFlags,
)
```
Changed to 

```zig
pub fn symLinkAbsoluteW(
    target_path_w: [:0]const u16,
    sym_link_path_w: [:0]const u16,
    flags: Dir.SymLinkFlags,
)
```

Or change the `windows.CreateSymbolicLink` function?